### PR TITLE
161966913 user reset password via mail 

### PR DESCRIPTION
**Pull request by kamyaD**
**What does this PR do?**
This PR allows the user to receive password reset notification on his e-mail and a link to reset the password.

**Description of Task to be completed?**
- send a  link to the user . for resetting password 
- reset the password through the link sent to the user

**How should this be manually tested?**
Clone the repo and cd into the project folder
Create a virtual environment by running $ python3 -m venv <name_of_environment>
Activate the virtual environment by running source <name_of_environment>/bin/activate
Run $ pip install -r requirements.txt
run python manage.py runserver

**What are the relevant pivotal tracker stories?**
[[#161966922]](https://www.pivotaltracker.com/n/projects/2225605)

**Screenshots:***
Passing tests:
screen shot 2018-12-17 at 06 26 10

Password Reset email notification:
screen shot 2018-12-17 at 06 29 43

### CODE REVIEW

File: authors/apps/authentication/templates/email_body.html [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            alt="Logo" style="width:50px;">
+        <h1 style="line-height: 35px;">
+            Hello <span id="username" style="color: #86C232;">{{username}}</span>
+        </h1>
+
+        <div id="div-content" style="line-height: 26px;">
+            <p> You requested for a password reset</p>
+            <p>Please click on the link bellow to reset your password</p>
+            <a id="confirm-link" href="{{link}}" style="background-color: #86C232; color:#fff; padding: 0.4em;font-size: 2em;text-decoration: none">CONFIRM
```
Comment (codeplus254) [inline authors/apps/authentication/templates/email_body.html]: Change this to reset password

**Comment by leonardnjura** [comment]
@kamyaD this PR is awesome. We badly needed this feature because who are we to deny this experience to our users? Everyone has a lot on their plate and often times we forget passwords.
Code climate, however, sniffed one or two things.

**Comment by r-wambui** [comment]
@kamyaD Please rebase develop and also work on merge conflicts

**Comment by kamyaD** [comment]
> @kamyaD Please rebase develop and also work on merge conflicts

@r-wambui  I managed to rebase develop and fixed the merge conflicts.

**Comment by kamyaD** [comment]
> @kamyaD this PR is awesome. We badly needed this feature because who are we to deny this experience to our users? Everyone has a lot on their plate and often times we forget passwords.
> Code climate, however, sniffed one or two things.

@leonardnjura Hi Leo, I managed to have a look at code climate issues and I think they are fixed now.

File: .travis.yml [hunk 2]
```diff
@@ -32,3 +32,6 @@ after_success:
 notifications:
   email: false
 
+# run coveralls after successful tests
```
Comment (r-wambui) [inline .travis.yml:35]: Why did you add this, it already in the travis.yml, did you pull the changes from develop?

File: authors/apps/authentication/serializers.py [hunk 3]
```diff
@@ -186,3 +188,20 @@ class SocialAuthSerializer(serializers.Serializer):
         max_length=2048, required=True, trim_whitespace=True)
     access_token_secret = serializers.CharField(
         max_length=2048, allow_null=True, default=None, trim_whitespace=True)
```
Comment (r-wambui) [inline authors/apps/authentication/serializers.py:190]: Please adhere to pep 8. Why don't we have blank lines?

File: authors/apps/authentication/urls.py [hunk 4]
```diff
@@ -12,4 +14,9 @@
     re_path(r'^user/activate/(?P<uidb64>[0-9A-Za-z_\-]+)/(?P<token>[a-zA-Z0-9\-_]+?\.[a-zA-Z0-9\-_]+?\.([a-zA-Z0-9\-_]+))/',
             ActivationView.as_view()),
     path('socialAuth/', SocialAuthAPIView.as_view()),
+    path('user/password-reset/', PasswordResetBymailAPIView.as_view()),
+    path('', views.index),
+    path('done', views.done),
```
Comment (r-wambui) [inline authors/apps/authentication/urls.py]: What is the purpose of this urls, `done`, `reset` and `thanks`

File: authors/apps/authentication/views.py [hunk 5]
```diff
@@ -17,15 +17,27 @@
...
 )
 from .models import User
+from rest_framework.generics import (RetrieveUpdateDestroyAPIView,
+    CreateAPIView,
+    UpdateAPIView)
+from django.core.mail import send_mail
+from ...settings import EMAIL_HOST_USER
+from .backends import JWTAuthentication
+from .renderers import UserJSONRenderer
+from .serializers import (LoginSerializer, RegistrationSerializer,
```
Comment (r-wambui) [inline authors/apps/authentication/views.py:46]: Why did you import serializers again and they already exist in line 20?

Comment (r-wambui) [reply authors/apps/authentication/views.py:46]: Also renderers, this exist in line 19 as well

**Comment by r-wambui** [comment]
Ensure the travis build runs.

**Comment by kamyaD** [comment]
> Ensure the travis build runs.

@r-wambui Hello Rose, I have worked on errors in Travis and now it runs, However, there are some failing tests whose features have not been worked on. I think that is why it's showing, "Some checks were not successful". What can we do about this? 

Please let me know if I did it as you expected.

**Comment by r-wambui** [comment]
@kamyaD Check that the tests passing in develop are also passing on this branch.

**Comment by ThaDeveloper** [comment]
It would be convenient to include the endpoint(s) for testing and documenting this PR's feature.

**Comment by kamyaD** [comment]
> @kamyaD Please rebase develop and also work on merge conflicts

I have done this

File: .travis.yml (hunk 2 above)
Comment (kamyaD) [reply .travis.yml:35]: This has been worked on

File: authors/apps/authentication/views.py (hunk 5 above)
Comment (kamyaD) [reply authors/apps/authentication/views.py:46]: This has been worked on

**Review by kamyaD — COMMENTED** [review]
> It would be convenient to include the endpoint(s) for testing and documenting this PR's feature.

I'm working on this

File: authors/apps/authentication/templates/email_body.html (hunk 1 above)
Comment (kamyaD) [reply authors/apps/authentication/templates/email_body.html]: This has been worked on
